src/components/Button/Button.tsx

```diff
@@ -1,78 +1,103 @@
-import React from 'react';
-import clsx from 'classnames';
+import React from 'react'
+import clsx from 'classnames'
 
-const filledBgColors: { [key: string]: string } = {
+type ButtonColor = {
+  [K in "green" | "indigo"]: string
+};
+
+const filledBgColors: ButtonColor = {
   green: 'bg-green-500',
   indigo: 'bg-indigo-500',
 };
 
-const filledBgHoverColors: { [key: string]: string } = {
+const filledBgHoverColors: ButtonColor = {
   green: 'hover:bg-green-700',
   indigo: 'hover:bg-indigo-700',
 };
 
-const filledBgFocusColors: { [key: string]: string } = {
+const filledBgFocusColors: ButtonColor = {
   green: 'focus:bg-green-400',
   indigo: 'focus:bg-indigo-400',
 };
 
-const filledBgActiveColors: { [key: string]: string } = {
+const filledBgActiveColors: ButtonColor = {
   green: 'active:bg-green-800',
   indigo: 'active:bg-indigo-800',
 };
 
-const filledShadowColors: { [key: string]: string } = {
+const filledShadowColors: ButtonColor = {
   green: 'shadow-md-green',
   indigo: 'shadow-md-indigo',
 };
 
-const filledShadowHoverColors: { [key: string]: string } = {
+const filledShadowHoverColors: ButtonColor = {
   green: 'hover:shadow-lg-green',
   indigo: 'hover:shadow-lg-indigo',
 };
 
-const outlineTextColors: { [key: string]: string } = {
+const outlineTextColors: ButtonColor = {
   green: 'text-black',
   indigo: 'text-indigo-500',
 };
 
-const outlineBorderColors: { [key: string]: string } = {
+const outlineBorderColors: ButtonColor = {
   green: 'border-green-500',
   indigo: 'border-indigo-500',
 };
 
-const outlineBgHoverColors: { [key: string]: string } = {
+const outlineBgHoverColors: ButtonColor = {
   green: 'hover:bg-green-50',
   indigo: 'hover:bg-indigo-50',
 };
 
-const outlineBorderHoverColors: { [key: string]: string } = {
+const outlineBorderHoverColors: ButtonColor = {
   green: 'hover:border-green-700',
   indigo: 'hover:border-indigo-700',
 };
 
-const outlineTextHoverColors: { [key: string]: string } = {
+const outlineTextHoverColors: ButtonColor = {
   green: 'hover:text-black',
   indigo: 'hover:text-indigo-700',
 };
 
-const outlineBgActiveColors: { [key: string]: string } = {
+const outlineBgActiveColors: ButtonColor = {
   green: 'active:bg-green-100',
   indigo: 'active:bg-indigo-100',
 };
 
-const Button: React.FC<{
+const initialClasses = 'flex items-center justify-center gap-1 font-bold outline-none tracking-wider focus:outline-none focus:shadow-none transition-all duration-300'
+
+const buttonSizes = {
```

Comment:
ano, toto riesenie je cistejsie riesenie

Stacilo by mi to slovne popisat, nebolo reba upravovat kod :)